### PR TITLE
Refactor engines utils. Delete the target

### PR DESCRIPTION
**Pull request by olpipi**
### Details:
 - Delete engines_test_util target
 - Move functions from engines_test_util to tests where they are used
 - Change namespaces to ov::tests::utils
 - Enhance ov::test::utils::copy_tensor_with_convert to not convert the same type

### Tickets:
 - [CVS-116636](https://jira.devtools.intel.com/browse/CVS-116636)

### CODE REVIEW

File: src/frontends/onnx/tests/onnx_utils.hpp [hunk 1]
```diff
@@ -6,4 +6,21 @@
...
+
 static const std::string ONNX_FE = "onnx";
+
+inline std::string backend_name_to_device(const std::string& backend_name) {
+    if (backend_name == "INTERPRETER")
+        return "TEMPLATE";
+    if (backend_name == "IE_CPU")
+        return "CPU";
+    if (backend_name == "IE_GPU")
+        return "GPU";
```
Comment (ilyachur) [inline src/frontends/onnx/tests/onnx_utils.hpp:19]: Can we remove these legacy names and use device names?

File: src/core/tests/control_dependencies.cpp [hunk 2]
```diff
@@ -53,6 +52,36 @@ class ControlDependencyOp : public ov::op::Op {
     }
 };
 
+::testing::AssertionResult test_ordered_ops(std::shared_ptr<ov::Model> f, const ov::NodeVector& required_ops) {
```
[the remark this answers is not in the document]
Comment (olpipi) [reply src/core/tests/control_dependencies.cpp]: done

File: src/core/tests/pass_manager.cpp [hunk 3]
```diff
@@ -19,6 +18,57 @@ using namespace std;
...
+    auto t0 = std::make_shared<ov::op::v1::Add>(arg_0, arg_1);
+    auto t1 = std::make_shared<ov::op::v0::MatMul>(t0, arg_2);
+    auto t2 = std::make_shared<ov::op::v1::Multiply>(t0, arg_3);
+
+    auto t3 = std::make_shared<ov::op::v1::Add>(t1, arg_4);
+    auto t4 = std::make_shared<ov::op::v1::Add>(t2, arg_5);
+
+    auto r0 = std::make_shared<ov::op::v1::Add>(t3, t4);
+
+    auto f0 = std::make_shared<ov::Model>(r0, ov::ParameterVector{arg_0, arg_1, arg_2, arg_3, arg_4, arg_5});
```
[the remark this answers is not in the document]
Comment (olpipi) [reply src/core/tests/pass_manager.cpp]: done

File: src/core/tests/utils/eval_utils.hpp [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+
+#include "gtest/gtest.h"
+#include "ngraph/runtime/host_tensor.hpp"
+#include "openvino/core/node.hpp"
+#include "openvino/core/shape.hpp"
+
+OPENVINO_SUPPRESS_DEPRECATED_START
+namespace {
+template <typename T>
+void copy_data(std::shared_ptr<ngraph::runtime::Tensor> tv, const std::vector<T>& data) {
```
[the remark this answers is not in the document]
Comment (olpipi) [reply src/core/tests/utils/eval_utils.hpp]: done

File: src/frontends/onnx/tests/onnx_editor.cpp [hunk 5]
```diff
@@ -593,7 +593,7 @@ OPENVINO_TEST(onnx_editor, subgraph__input_edge_from_tensor_with_multiple_consum
                                      "onnx/model_editor/reference/"
                                      "subgraph__input_edge_from_tensor_with_multiple_consumers_5.onnx");
 
-    const auto result = compare_onnx_models(editor.model_string(), ref_model);
+    const auto result = ngraph::test::compare_onnx_models(editor.model_string(), ref_model);
```
[the remark this answers is not in the document]
Comment (olpipi) [reply src/frontends/onnx/tests/onnx_editor.cpp]: deleted

File: src/frontends/onnx/tests/onnx_utils.hpp (hunk 1 above)
Comment (olpipi) [reply src/frontends/onnx/tests/onnx_utils.hpp]: done

Comment (olpipi) [reply src/frontends/onnx/tests/onnx_utils.hpp]: I changed hardcoded valued to variables from ov::test::utils.
To get rid of INTERPRETER, IE_CPU and IE_GPU names, I need to change more deeply CMakeLists.txt and test manifest files.
So, I'd prefer to do it in scope of the next PR

File: src/core/tests/utils/eval_utils.hpp [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+    } else if (et == ov::element::u8) {
+        init_int_tv<uint8_t>(tv, engine, 0, 1);
+    } else if (et == ov::element::u16) {
+        init_int_tv<uint16_t>(tv, engine, 0, 1);
+    } else if (et == ov::element::u32) {
+        init_int_tv<uint32_t>(tv, engine, 0, 1);
+    } else if (et == ov::element::u64) {
+        init_int_tv<uint64_t>(tv, engine, 0, 1);
+    } else {
+        OPENVINO_THROW("unsupported type");
```
Comment (p-durandin) [inline src/core/tests/utils/eval_utils.hpp:116]: Worse to move to switch?

**Review by p-durandin — APPROVED** [review]
LGTM

**Review by ilyachur — APPROVED** [review]

**Pull request merged**